src/microbe_stage/editor/MicrobeEditor.cs

```diff
@@ -372,6 +379,16 @@ public void OnFinishLoading(Save save)
         StartMusic();
     }
 
+    public void OnFinishTransitioning()
+    {
+        transitionFinished = true;
+
+        if (!IsLoadedFromSave)
+        {
+            InitEditorFresh();
+        }
```

Comment:
I don't think this is the right way either.

Reply:
When the transition finishes, I thought we can call InitEditorFresh again to actually execute the whole method? Since if either auto evo or transition hasn't finished, ready would still be false and EditorReady is not called.

Reply:
That method does other setup as well, which I want to run before the fade-in works.
Look at closely what it does, it does a bunch of setup even when it detects that auto-evo isn't ready.

Reply:
InitEditorFresh is also called by OnEnterEditor which is called in _Ready, so wouldn't the setups in InitEditorFresh run just fine? The setups would run multiple times though, considering how InitEditor is called in OnEnterEditor, OnFinishLoading, and OnFinishTransitioning.

Reply:
No if you have the current code where you bail out of `InitEditorFresh` if transition is not finished.
This wasn't even my main comment about this...

https://github.com/Revolutionary-Games/Thrive/pull/1590#discussion_r485507994 is what I actually want, InitEditorFresh to only skip the `OnEditorReady` call if transition is still in progress.

Reply:
Oooh ok I see. But what if the fade transition finishes after both calls to InitEditorFresh (since InitEditor is called in two places I assumed that InitEditorFresh would be called twice in total) so that OnEditorReady is never reached in the if statement? Or does that never actually happen?

Reply:
The other place where InitEditor is called, when a save is loaded. It should never be the case that InitEditor is called twice.
The logic is that when loading the editor attempt to call InitEditor, it does some setup always, and if both auto-evo and the fade is ready then it calls editor ready. If editor ready has not been called the _Process in the editor checks if auto-evo and fade are ready now, if they are then on ready is called.

Reply:
Right I see. _Process method takes care of the fade not being finished on time issue.